src/Build/Instance/HostServices.cs

```diff
@@ -52,6 +56,9 @@ public class HostServices
         /// </summary>
         private Dictionary<string, NodeAffinity> _projectAffinities;
 
+#if FEATURE_COM_INTEROP
+        private IRunningObjectTableWrapper _runningObjectTable = new RunningObjectTable();
```

Comment:
A question/discussion on Dispose:

You're not calling it on the RunningObjectTable, and in the CPS code I am. @davkean expressed concern about disposing because it calls Marshal.ReleaseComObject which is a bad thing to do. Looking at [the documentation](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.marshal.releasecomobject?view=netframework-4.7.2) it doesn't seem bad to me (it just decrements the ref count, it doesn't ignore it like FinalReleaseComObject) but it also says to use it "only if it is absolutely required".

I'm guessing since the ROT is a singleton none of it matters, but figured we should at least do the same thing in both repos. So no disposal?

Reply:
If there are 10 references to a COM on the .NET side, the ref count is only ever 1. When one .NET object calls ReleaseComObject, it's saying "no other .NET code uses this reference". It's better just to let the GC handle this.

Reply:
Thank you. That's not apparent to me from reading the doco, but it makes sense. I'll remove the disposals on my side :)

Reply:
I would delete it so that no one is tempted to call it.

Reply:
I'm never sure with these things that get copied around to every repository - do you modify them to keep only the bits you call? Or leave them the same so one day someone can find all the dupes and clean them up

Reply:
The usual approach in this repo is to aim for minimal diffs, for ease of future maintenance--hopefully a diff can apply cleanly to the various different copies. But it's not a firm rule and I don't know which is better in this particular case.

Reply:
i'll remove Dispose method as well.

Reply:
Just to be clear: I mean I removed the _calls_ to Dispose in the CPS code (which you already don't have). I haven't changed the definition of the RunningObjectTable class

Reply:
ah, ok, i won't change that

Reply:
that means GC will call Dispose at the correct time?

Reply:
The ref count will go down *at some point*, but there isn't a guarantee of when.  If you need it to happen at a specific time, use [`Marshal.GetUniqueObjectForIUnknown`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.marshal.getuniqueobjectforiunknown?view=netframework-4.7.2#System_Runtime_InteropServices_Marshal_GetUniqueObjectForIUnknown_System_IntPtr_), and then `Marshal.ReleaseComObject` on that.  We've had to do that for the Invisible editor, since it saves when it's ref count goes to 0...

Reply:
GC does not call Dispose nor know about Dispose. GC knows about Runtime Callable Wrappers (RCW), and will clean them up when they fall out of scope and have no managed references.